arch/src/x86_64/regs.rs

```diff
@@ -239,6 +246,11 @@ fn setup_page_tables(mem: &GuestMemoryMmap, sregs: &mut kvm_sregs) -> Result<()>
     sregs.cr3 = PML4_START.raw_value();
     sregs.cr4 |= X86_CR4_PAE;
     sregs.cr0 |= X86_CR0_PG;
+
+    if CpuidPatch::get_cpuid_reg(cpuid, 0x7, None, CpuidReg::ECX) == 0x10000 {
```

Comment:
I don't think getting it from the patch is right. Also you're checking if only one bit is set:

I think this is better
```rust
if unsafe { std::arch::x86_64::__cpuid(7).ecx } & (1 << 16) != 0 {
         sregs.cr4 |= X86_CR4_LA57; 
}
```